impress-org/givewp
# Donor Profile authentication logic is now implemented

Resolves #5567

## Description
This PR implements backend and frontend logic to support authentication in the Donor Profile application. Previously, the app would break when viewed by donors who were not logged in. Now, when not logged in, donors are prompted to log in or enter an email address to create an account.

## Affects
This PR affects the REST routes available for the Donor Profile domain (adding Login, Logout) and the frontend rendering logic, requiring that donors log in to view their profile.

## Visuals
<img width="970" alt="Screen Shot 2021-01-26 at 2 23 41 PM" src="https://user-images.githubusercontent.com/5186078/105894680-885d5e00-5fe2-11eb-9bc2-2b4b8aa28f32.png">

## Pre-review Checklist


## Testing Instructions
1. Add a new page with the Donor Profile block
2. View the page from the frontend, logged in. Check that Profile functions as expected (can modify profile, view donation history, etc).
3. Logout, and view the page. Check that you are prompted to login, and that login fields work as expected.
